NumericOutcomeCompression.md

```diff
@@ -101,261 +101,265 @@ these integers (interpreted in base `B`) where all digits not used may be any va
-corresponds to the number of unique digits between the start and end of the interval and for each unique digit a row is
-generated in both the front and back groupings of length at most `B-1 ` which corresponds to the coefficient in the order bound.
+The worst case interval yielding the highest number of adaptor signatures to be computed in a base `B` with `n` digits is the interval `[1, n - 2]` for which one need to compute `(B * (n - 1) * 2) + (n - 2)`.
+In the worst case, base 2 is thus optimal.
+This also appears to be the case in general.
 
 The total number of digits for which intermediary adaptor point must be generated to cover an interval of length `L` made of `n` digits will be `n*L`.
 Since the number of digits is smaller for larger bases (e.g. the number 100 is three digits in base 10 but seven in base 2), larger bases might at first sight seem more optimal, even though they require a larger number of adaptor signatures.
 However, applying the [pre-computation optimization](#pre-computing), we can reduce the number of intermediary anticipation points to compute to `B*floor(log_B(n) + 1)`, which is minimal for `B = 2`.
 Base 2 thus both requires computing less adaptor signatures and less intermediary anticipation points, and is thus chosen as the preferred one.
 
-#### Optimizations
-
-Because `start` and `end` are outliers to the general grouping pattern, there are optimizations that could potentially be made when they are added.
-
-Consider the example in base 10 of the interval `[2200, 4999]` which has the endpoints `2200` and `4999` along with the groupings
-
-```
-2201, 2202, 2203, 2204, 2205, 2206, 2207, 2208, 2209,
-221_, 222_, 223_, 224_, 225_, 226_, 227_, 228_, 229_,
-23__, 24__, 25__, 26__, 27__, 28__, 29__,
+### Algorithms
 
-3___,
+Computing the prefixes required to cover the intervals with constant payouts can be done in many ways.
+Here we provide two examples, a recursive version that might facilitate the understanding, and an iterative one that can be used in practice if recursion is undesirable.
 
-40__, 41__, 42__, 43__, 44__, 45__, 46__, 47__, 48__,
-490_, 491_, 492_, 493_, 494_, 495_, 496_, 497_, 498_,
-4990, 4991, 4992, 4993, 4994, 4995, 4996, 4997, 4998
-```
+We first define a function to decompose any number into its digits in a given base.
 
-This grouping pattern captures the exclusive interval `(2200, 4999)` and then adds the endpoints in ad-hoc to get the inclusive range `[2200, 4999]`.
-But this method misses out on a good amount of compression as re-introducing the endpoints allows us to replace the first two rows with
-a single `22__` and the last 3 rows with just `4___`.
+```rust
+fn decompose_value(mut value: usize, base: usize, nb_digits: usize) -> Vec<usize> {
+    let mut res = Vec::new();
 
-This optimization is called the **endpoint optimization**.
+    while value > 0 {
+        res.push(value % base);
+        value = ((value as f64) / (base as f64)).floor() as usize;
```

Comment:
maybe should just make `value` and `base` `f64`s insted of casting them since it is just an example

Reply:
Mmmmm I'm a bit hesitant because I feel it's quite important to return integers and not floats and I'd be worried that someone just copy pasting that thinks that they can just manipulate floats all over and be fine (tbh I'm not sure if it's actually fine or not, but I think it's just better to avoid it).